src/libraries/src.proj

```diff
@@ -7,8 +7,8 @@
   <ItemGroup>
     <_allSrc Include="$(MSBuildThisFileDirectory)*\src\*.csproj"
              Exclude="@(ProjectExclusions)" />
-    <!-- Include the System.Text.Json.SourceGeneration project -->
-    <_allSrc Include="$(MSBuildThisFileDirectory)System.Text.Json\source_generation\src\System.Text.Json.SourceGeneration.csproj" />
+    <NonNetCoreAppProject Include="$(MSBuildThisFileDirectory)*\gen\*.csproj"
```

Comment:
Should we create a top-level project instead of having a sub-project?

Reply:
It's tied close to the library and will ship in the library's package as another asset type.  IMHO we break more of the dotnet/runtime conventions by lifting it up (no ref assembly, src implies runtime assembly, which is not the case)  I think we need a new consistent convention for library generators.  Right now the proposed convention is the "gen" folder.

Reply:
It looks to me like you don't follow this convention.  You had sub-directories under `gen`.

Reply:
I've updated `gen` to have only the source for the generator; and moved the tests to the root `tests` directory to sit beside the existing JSON tests. This is what the logging generator does - https://github.com/dotnet/runtime/tree/main/src/libraries/Microsoft.Extensions.Logging.